larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -273,11 +271,9 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
 
   // Now allocate the buffers
   if (need_im2col) {
-    // Allocate the im2col tensor if necessary
-    if (op_data->im2col_id == kTensorNotAllocated) {
```

Comment:
With this if statement things did no longer work, so I had to remove it.

Reply:
I'm still a bit confused by this, they still seem to have this check in the latest conv kernel code: https://github.com/tensorflow/tensorflow/blob/bdb6f62c32ee16016de8ec83e42a96b59ac4f9fa/tensorflow/lite/kernels/conv.cc#L272

My worry is that if `Prepare` is now called several times, after resizing the input for example, that it allocates a new tensor without de-allocating the old one.

Reply:
I can have another look, I'll revert the fix, re-run the failing test, and try to get more debug info.

Reply:
This should be fixed now, in the end all we needed was the temporaries assignment to move outside, not the add-tensors.